plugin/core/views.py

```diff
@@ -425,7 +425,15 @@ def text_document_identifier(view_or_uri: Union[DocumentUri, sublime.View]) -> T
     return {"uri": uri}
 
 
+def get_first_selection_region(selelection: sublime.Selection) -> Optional[sublime.Region]:
+    try:
+        return selelection[0]
+    except IndexError:
+        return None
+
+
 def first_selection_region(view: sublime.View) -> Optional[sublime.Region]:
+    """@deprecated Use :meth:`get_first_selection_region` instead."""
```

Comment:
Hm... what's wrong with this function to deprecate it?

If I had written this function, I probably would not have used a `try ... except` for it, but instead
```python
selection = view.sel()
return selection[0] if selection else None
```
... to save a few lines, but that's just a matter of preference.

Reply:
I think there is a bug in 3.3 host that makes it evaluate to true when not empty. That is why try catch is used

Reply:
As far as deprecation. Well, decide to switch to it everywhere instead of having two similar ones but then noticed that is used in a couple of helper packages

Reply:
> I think there is a bug in 3.3 host that makes it evaluate to true when not empty.

The result of `View.sel()` is a list, so I would expect it to evaluate to true in all cases except if there is no cursor in the view (only possible via plugins). Or do you have more information about this "bug"?

Reply:
Looking at the API docs, I think
```python
return selection[0] if len(selection) else None
```
should do the trick. But not necessary to change it here.